protocols/kad/src/kbucket/key.rs

```diff
@@ -167,7 +167,14 @@ impl AsRef<KeyBytes> for KeyBytes {
 
 /// A distance between two keys in the DHT keyspace.
 #[derive(Copy, Clone, PartialEq, Eq, Default, PartialOrd, Ord, Debug)]
-pub struct Distance(pub U256);
+pub struct Distance(pub(super) U256);
+
+impl Distance {
+    /// Returns the base 2 logarithm of the [`Distance`].
+    pub fn log2(&self) -> Option<u32> {
+        (256 - self.0.leading_zeros()).checked_sub(1)
```

Comment:
```suggestion
    /// Returns the integer part of the base 2 logarithm of the [`Distance`].
    pub fn ilog2(&self) -> Option<u32> {
        (256 - self.0.leading_zeros()).checked_sub(1)
```
Just for added clarity.